bbq/compiler/compiler.go

```diff
@@ -1425,6 +1425,11 @@ func (c *Compiler[_, _]) VisitInvocationExpression(expression *ast.InvocationExp
 
 		// Receiver is loaded first. So 'self' is always the zero-th argument.
 		c.compileExpression(invokedExpr.Expression)
+
+		if _, ok := memberInfo.AccessedType.(*sema.ReferenceType); ok {
+			c.codeGen.Emit(opcode.InstructionDeref{})
+		}
```

Comment:
Would it be better to do this at runtime, like `opInvokeDynamic` used to be? Wonder if the accessed-type could be a non-reference at compile-time, but the runtime-value be a reference. For e.g: `getType()` on an `AnyStruct` variable, where the actual value is a reference.

Reply:
Also, should we handle optional here?